src/actions/receive-profile.js

```diff
@@ -1051,6 +1051,26 @@ export function retrieveProfileFromBrowser(
       });
       const unpackedProfile =
         await _unpackGeckoProfileFromBrowser(rawGeckoProfile);
+      const meta = unpackedProfile.meta;
+      if (meta.configuration.features.includes('power')) {
+        const tracks = await browserConnection.getExternalPowerTracks(
+          meta.startTime + meta.profilingStartTime,
+          meta.startTime + meta.profilingEndTime
+        );
+        for (const track of tracks) {
+          const samples = track.sample_groups[0].samples;
+          const timeColumnIndex = samples.schema.time;
+          for (const sample of samples.data) {
+            // Adjust the sample times to be relative to meta.startTime,
+            // and limit the precision to nanoseconds
+            sample[timeColumnIndex] =
+              Math.round(
+                (sample[timeColumnIndex] + meta.profilingStartTime) * 1e6
+              ) / 1e6;
+          }
+          unpackedProfile.counters.push(track);
+        }
+      }
```

Comment:
I feel like we can extract this code to a function that we can put inside `src/profile-logic/process-profile.js`. What do you think?